test/e2e/specs/editor/various/block-bindings.spec.js

```diff
@@ -0,0 +1,856 @@
+		await requestUtils.activateTheme( 'emptytheme' );
+		await requestUtils.activatePlugin( 'gutenberg-test-block-bindings' );
+		await requestUtils.deleteAllMedia();
+		const placeholderMedia = await requestUtils.uploadMedia(
+			path.join( './test/e2e/assets', '10x10_e2e_test_image_z9T8jK.png' )
+		);
+		variables.placeholderSrc = placeholderMedia.source_url;
+		// Init image blocks.
+		variables.blocks.images.urlOnly = {
+			name: 'core/image',
+			attributes: {
+				url: variables.placeholderSrc,
+				alt: 'default alt value',
+				title: 'default title value',
+				metadata: {
+					bindings: {
+						url: {
+							source: 'core/post-meta',
+							args: { key: 'url_custom_field' },
+						},
+					},
+				},
+			},
+		};
+		variables.blocks.images.altOnly = {
+			name: 'core/image',
+			attributes: {
+				url: variables.placeholderSrc,
+				alt: 'default alt value',
+				title: 'default title value',
+				metadata: {
+					bindings: {
+						alt: {
+							source: 'core/post-meta',
+							args: { key: 'text_custom_field' },
+						},
+					},
+				},
+			},
+		};
+		variables.blocks.images.titleOnly = {
+			name: 'core/image',
+			attributes: {
+				url: variables.placeholderSrc,
+				alt: 'default alt value',
+				title: 'default title value',
+				metadata: {
+					bindings: {
+						title: {
+							source: 'core/post-meta',
+							args: { key: 'text_custom_field' },
+						},
+					},
+				},
+			},
+		};
+		variables.blocks.images.multipleAttrs = {
+			name: 'core/image',
+			attributes: {
+				url: variables.placeholderSrc,
+				alt: 'default alt value',
+				title: 'default title value',
+				metadata: {
+					bindings: {
+						url: {
+							source: 'core/post-meta',
+							args: { key: 'url_custom_field' },
+						},
+						alt: {
+							source: 'core/post-meta',
+							args: { key: 'text_custom_field' },
+						},
+					},
+				},
+			},
+		};
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllPosts();
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllMedia();
+		await requestUtils.activateTheme( 'twentytwentyone' );
+		await requestUtils.deactivatePlugin( 'gutenberg-test-block-bindings' );
+	} );
+
+	test.describe( 'Template context', () => {
+		test.beforeEach( async ( { admin, editor } ) => {
+			await admin.visitSiteEditor( {
+				postId: 'emptytheme//index',
+				postType: 'wp_template',
+			} );
+			await editor.canvas.locator( 'body' ).click();
+			await editor.openDocumentSettingsSidebar();
+		} );
+
+		test.describe( 'Paragraph', () => {
+			test( 'Should show the value of the custom field', async ( {
+				editor,
+			} ) => {
+				await editor.insertBlock( variables.blocks.paragraph );
+				const paragraphBlock = editor.canvas.getByRole( 'document', {
+					name: 'Block: Paragraph',
+				} );
+				const paragraphContent = await paragraphBlock.textContent();
+				expect( paragraphContent ).toBe(
+					variables.customFields.textKey
+				);
```

Comment:
There's a handy [`toHaveText`](https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-have-text) assertion ;)

Reply:
I've used `toHaveText` in this commit as part of a follow-up pull request: [link](https://github.com/WordPress/gutenberg/commit/34a2255f746ad040baa9181a68e185374b9d752d).